core/src/heaviest_subtree_fork_choice.rs

```diff
@@ -498,6 +498,30 @@ impl HeaviestSubtreeForkChoice {
         );
     }
 
+    fn heaviest_slot_on_same_voted_fork(&self, tower: &Tower) -> Option<Slot> {
+        tower
+            .last_voted_slot()
+            .map(|last_voted_slot| {
+                let heaviest_slot_on_same_voted_fork = self.best_slot(last_voted_slot);
+                if heaviest_slot_on_same_voted_fork.is_none() && tower.is_stray_last_vote() {
+                    info!("returning None because of stray last vote...");
+                    // return None because bank_forks doesn't have corresponding banks
+                    return None;
+                };
+                let heaviest_slot_on_same_voted_fork = heaviest_slot_on_same_voted_fork.expect(
```

Comment:
Similar to this [reasoning](https://github.com/solana-labs/solana/pull/10718/files#r455530911), I retained `.expect()` here. Btw, these code wasn't existing at #9902.